source/unix/openssl_platform_init.c

```diff
@@ -583,6 +634,8 @@ static enum aws_libcrypto_version s_resolve_libcrypto(void) {
         result = s_resolve_libcrypto_lib();
     }
 
+    s_validate_libcrypto_linkage();
```

Comment:
Another call-out for reviewers -- if we're concerned about this affecting downstream consumers that have have cross-linked libcryptos, we can consider making this check an opt-in CMake flag. Perhaps something like `-DVALIDATE_LIBCRYPTO_LINK_VERSION=ON`.

Reply:
Fair call-out, but for simplicity's sake let's have it on by default